### DISCUSSION TITLE
How to Select a column with a @Query

### DISCUSSION THREAD

**Discussion by impakte** [vitusortner/floor#677] [Q&A] [answered]
Hi everyone, 

I'm a new user of Floor and I would like to know if there is a way to do a Query to Select a Column 

Example: 
My Table Product has in columns:  Name, Barecode, Price

I would like to get each Barecode so in my ProductDao I did: 

@Query('SELECT barecode FROM $stocksSqliteTableName')
  Future<List<String?>> getAllBarecode();

when I generate my database.g file, the following error appears: 
The body might complete normally, causing 'null' to be returned, but the return type, 'FutureOr<List<String?>>', is a potentially non-nullable type.
Try adding either a return or a throw statement at the end.

Thank's for your help

**Comment by acuba001** [comment, marked as answer]
I'm no expert, but I believe that a select view would be able to do the trick. Apparently floor supports it: https://pinchbv.github.io/floor/database-views/

Hope this helps.

**Reply by dkaera** [reply]
to @impakte 
Yeah, @acuba001 is right, DatabaseViews helps to represent data by specified columns but requires some effort to implement a DatabaseView DAO. Note that DatabaseView does not support streams, but it fully covers use cases for representing table data in any form :)

**Comment by dkaera** [comment]
Hey @impakte @acuba001 

Since Floor is an ORM framework, we cannot access a specific column. To successfully generate `Query`, we need to use the corresponding Entity model and add a mapping method to retrieve a specific field from the Entity model.

In your case it should look as a sample below:

```
@dao
abstract class ProductDao {
  @Query('SELECT * FROM product')
  Future<List<Product>> getProducts();

  Future<List<String>> getBarcodes() => getProducts()
      .then((products) => products.map((product) => product.barcode).toList());
}
```

**Reply by dkaera** [reply]
This is a simple alternative to DatabaseView.
